src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx

```diff
@@ -169,72 +167,192 @@ export const data = defineData({
 
-- **API.ts** – All the type definitions for queries, mutations, and subscriptions
-- **mutations.ts** – The GraphQL mutations from your API
-- **queries.ts** – The GraphQL queries from your API
-- **subscriptions.ts** – The GraphQL subscriptions from your API
+The supported data sources are:
+- Amazon DynamoDB
+- AWS Lambda
+- Amazon RDS databases with Data API
+- Amazon EventBridge
+- OpenSearch
+- HTTP endpoints
 
-To call the `echo` query from step 1, we need to import the `echo` function from **queries.ts** and pass it as a query to `client.graphql(...)`.
+You can add these additional data sources via our the `amplify/backend.ts` file:
 
-```ts
-import { generateClient } from 'aws-amplify/data';
-import { type Schema } from '@/amplify/data/resource';
-import * as queries from '@/graphql/queries';
+```ts title="amplify/backend.ts"
+import * as dynamoDb from 'aws-cdk-lib/aws-dynamodb'
+import { defineBackend } from '@aws-amplify/backend';
+import { auth } from './auth/resource.js';
+import { data } from './data/resource.js';
+
+export const backend = defineBackend({
+  auth,
+  data,
+});
+
+const externalDataSourcesStack = backend.createStack("MyExternalDataSources")
+
+const externalTable = dynamoDb.Table.fromTableName(externalDataSourcesStack, "MyTable", "MyExternalTable")
+
+backend.data.addDynamoDbDataSource(
+  // highlight-next-line
+  "ExternalTableDataSource",
+  externalTable)
+```
 
-const client = generateClient<Schema>();
+In your schema you can then reference these additional data sources based on their name:
 
-async function callEcho() {
-  const response = await client.graphql({
-    query: echo,
-    variables: {
-      content: 'Echo me!'
+```ts title="amplify/data/resource.ts"
+import {
+  type ClientSchema,
+  a,
+  defineData,
+} from '@aws-amplify/backend';
+
+const schema = a.schema({
+  Post: a.model({
+    content: a.string(),
+    likes: a.integer()
+      .authorization([a.allow.private().to(['read'])])
+  }).authorization([
+    a.allow.owner(),
+    a.allow.private().to(['read'])
+  ]),
+
+  likePost: a
+    .mutation()
+    .arguments({ postId: a.id() })
+    .returns(a.ref('Post'))
+    .authorization([a.allow.private()])
+    .handler(a.handler.custom({
+      // highlight-next-line
+      dataSource: "ExternalTableDataSource",
+      entry: './increment-like.js'
+    }))
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: 'apiKey',
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30
     }
-  });
+  },
+});
+```
 
-  console.log(response.data.echo?.content);
-  console.log(response.data.echo?.executionDuration);
-}
+</Block>
+</BlockSwitcher>
+
+<Callout warning>
+
+The `.handler()` accepts an array of handlers that will run in a pipeline. For now, in the developer, you can only use `a.handler.custom()` all other handlers, such as `a.handler.function()` are still under active development. 
```

Comment:
did we mean to say `developer preview` here.
this sentence reads a bit odd

Reply:
```suggestion
The `.handler()` accepts an array of handlers that will run in a pipeline. For now, in the developer preview, you can only use `a.handler.custom()`. All other handlers, such as `a.handler.function()` are still under active development. 
```